src/status_im/contexts/wallet/wallet_connect/events/session_requests.cljs

```diff
@@ -61,14 +62,17 @@
  :wallet-connect/process-personal-sign
  (fn [{:keys [db]}]
    (let [[raw-data address] (data-store/get-db-current-request-params db)
-         parsed-data        (native-module/hex-to-utf8 raw-data)]
+         hex-message        (if (utils-address/has-hex-prefix? raw-data)
+                              raw-data
+                              (native-module/utf8-to-hex raw-data))
+         parsed-data        (native-module/hex-to-utf8 hex-message)]
```

Comment:
Looks like if `hex-message` is false, we will make utf8-to-hex conversion and than back :)
can it be replaced with?
```clojure
parsed-data (if (utils-address/has-hex-prefix? raw-data)
                    (native-module/hex-to-utf8 raw-data)
                    raw-data)
```